src/GTP.cpp

```diff
@@ -243,6 +243,146 @@ std::string GTP::get_life_list(const GameState & game, bool live) {
+            if (cmdstream.fail()) {
+                break;
+            }
+
+            std::vector<int> moves;
+            std::istringstream movestream(textmoves);
+            while (!movestream.eof()) {
+                std::string textmove;
+                getline(movestream, textmove, ',');
+                int sepidx = textmove.find_first_of(':');
+                if (sepidx != -1) {
+                    if (!(sepidx == 2 || sepidx == 3)) {
+                        moves.clear();
+                        break;
+                    }
+                    auto move1_compressed = game.board.text_to_move(
+                        textmove.substr(0, sepidx)
+                    );
+                    auto move2_compressed = game.board.text_to_move(
+                        textmove.substr(sepidx + 1)
+                    );
+                    if (move1_compressed == FastBoard::NO_VERTEX ||
+                        move1_compressed == FastBoard::PASS ||
+                        move1_compressed == FastBoard::RESIGN ||
+                        move2_compressed == FastBoard::NO_VERTEX ||
+                        move2_compressed == FastBoard::PASS ||
+                        move2_compressed == FastBoard::RESIGN)
+                    {
+                        moves.clear();
+                        break;
+                    }
+                    auto move1_xy = game.board.get_xy(move1_compressed);
+                    auto move2_xy = game.board.get_xy(move2_compressed);
+                    int xmin = std::min(move1_xy.first, move2_xy.first);
+                    int xmax = std::max(move1_xy.first, move2_xy.first);
+                    int ymin = std::min(move1_xy.second, move2_xy.second);
+                    int ymax = std::max(move1_xy.second, move2_xy.second);
+                    for (int move_x = xmin; move_x <= xmax; move_x++) {
+                        for (int move_y = ymin; move_y <= ymax; move_y++) {
+                            moves.push_back(game.board.get_vertex(move_x,move_y));
+                        }
+                    }
+                } else {
+                    auto move = game.board.text_to_move(textmove);
+                    if (move == FastBoard::NO_VERTEX) {
+                        moves.clear();
+                        break;
+                    }
+                    moves.push_back(move);
+                }
+            }
+            if (moves.empty()) {
+                break;
+            }
+
+            int color;
+            if (textcolor == "w" || textcolor == "white") {
+                color = FastBoard::WHITE;
+            } else if (textcolor == "b" || textcolor == "black") {
+                color = FastBoard::BLACK;
+            } else {
+                break;
+            }
+
+            if (until_movenum < 1) {
+                break;
+            }
+            until_movenum += game.get_movenum() - 1;
+
+            for (auto move : moves) {
```

Comment:
This can take a (const) reference.

Reply:
Is this because you prefer it from a coding-style point of view? I think for plain int vectors it would not make much of a difference.